jujugui/static/gui/src/app/app.js

```diff
@@ -760,13 +760,17 @@ YUI.add('juju-gui', function(Y) {
         how to render.
     */
     _renderInspector: function(metadata) {
+      var service = this.db.services.getById(metadata.id);
       React.render(
         <components.Panel
           instanceName="inspector-panel"
           visible={true}
           metadata={metadata}>
-          <components.Inspector>
-            <components.ServiceOverview />
+          <components.Inspector
+            service={service}
+            changeState={this.changeState.bind(this)}>
+            <components.ServiceOverview
+              service={service} />
```

Comment:
If that ends up being the case then the inspector can pass the service model to the child.